platform/persistence/platform-persistence-core-attributes-impl/src/main/java/org/codice/ddf/persistence/attributes/internal/AttributesStoreImpl.java

```diff
@@ -0,0 +1,136 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p/>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p/>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+
+package org.codice.ddf.persistence.attributes.internal;
+
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+
+import org.apache.commons.lang.StringUtils;
+import org.codice.ddf.persistence.PersistenceException;
+import org.codice.ddf.persistence.PersistentItem;
+import org.codice.ddf.persistence.PersistentStore;
+import org.codice.ddf.persistence.attributes.AttributesStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class AttributesStoreImpl implements AttributesStore {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(AttributesStoreImpl.class);
+
+    private PersistentStore persistentStore;
+
+    private ReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+
+    private static final String EMPTY_USERNAME_ERROR = "Empty username specified";
+
+    public AttributesStoreImpl(PersistentStore persistentStore) {
+        this.persistentStore = persistentStore;
+    }
+
+    @Override
+    public long getCurrentDataUsageByUser(final String username) throws PersistenceException {
+        long currentDataUsage = 0L;
+
+        if (StringUtils.isEmpty(username)) {
+            throw new PersistenceException(EMPTY_USERNAME_ERROR);
+        }
+
+        try {
+            readWriteLock.readLock()
+                    .lock();
+            currentDataUsage = getCurrentDataUsageByUserNoLock(username);
+        } finally {
+            readWriteLock.readLock()
+                    .unlock();
+        }
+
+        return currentDataUsage;
+    }
+
+    @Override
+    public void updateUserDataUsage(final String username, long newDataUsage)
```

Comment:
It's a bit confusing that only one of these parameters is `final` when both could be. Same goes for `setDataUsage`.